src/scripts/storage/clipperData.ts

```diff
@@ -0,0 +1,49 @@
+import {GetResponseAsync, TimeStampedData} from "../http/cachedHttp";
+import {ClipperCachedHttp} from "../http/clipperCachedHttp";
+
+import {Logger} from "../logging/logger";
+
+import {ClipperStorageGateStrategy} from "./clipperStorageGateStrategy";
+import {Storage} from "./storage";
+import {StorageGateStrategy} from "./StorageGateStrategy";
+
+/**
+ * The data-access-like object that handles the fetching and caching of data
+ * from Clipper-specific endpoints (TODO: a factory class that is responsible for
+ * providing getRemoteValue objects?). Also provides a gating mechanism as a
+ * sanity-check for what information gets stored locally.
+ */
+export class ClipperData implements Storage {
+	private storage: Storage;
+	private storageGateStrategy: StorageGateStrategy;
+	private cachedHttp: ClipperCachedHttp;
+
+	constructor(storage: Storage, logger?: Logger) {
+		this.storage = storage;
+		this.storageGateStrategy = new ClipperStorageGateStrategy(storage);
```

Comment:
As I understand it, `clipperData.ts` has a single strategy. This strategy is the concrete `ClipperStorageGateStrategy` which gates 2 keys based on `userInfo`'s presence. Is there a way we can reverse the interface, so we can inject strategies into the `clipperData`? Going a step further, we can specify a strategy on a per key basis? It might not be worth it, but if we need to add another gating strategy to something in ClipperData I think it would make it easier. Also it would make testing a little easier imo (although I've looked at your tests and they are very complete).

I am envisioning injecting `clipperData` with a map of keys to strategies (`{[string]: StorageGateStrategy}` or `StrategyMap[]` where `StrategyMap = {[string]: StorageGateStrategy`) or possibly having a method to `setStrategies` for specific keys (`ClipperData::setStrategy(keys: string[], strategy: StorageGateStrategy)`).  ClipperData can check the strategy (if one exists) for the key passed into setValue.

Even if we don't end up implementing it like this, I wonder what the pros/cons of this method versus the current implementation would be.